apps/bixarena/api/src/main/java/org/sagebionetworks/bixarena/api/service/PublicStatsService.java

```diff
@@ -0,0 +1,49 @@
+package org.sagebionetworks.bixarena.api.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.sagebionetworks.bixarena.api.configuration.CacheNames;
+import org.sagebionetworks.bixarena.api.model.dto.PublicStatsDto;
+import org.sagebionetworks.bixarena.api.model.repository.BattleRepository;
+import org.sagebionetworks.bixarena.api.model.repository.UserStatsRepository;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service for retrieving public platform statistics.
+ * Results are cached to reduce database load.
+ */
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class PublicStatsService {
+
+  private final UserStatsRepository userStatsRepository;
+  private final BattleRepository battleRepository;
+
+  /**
+   * Get public platform statistics.
+   * This method is cacheable and returns the DTO object directly.
+   *
+   * @return Public statistics DTO
+   */
+  @Cacheable(value = CacheNames.PUBLIC_STATS, key = "'" + CacheNames.PUBLIC_STATS_KEY + "'")
```

Comment:
Two notes: 1) `@Cacheable` only works on public method, and 2) we cache here on this method that returns `PublicStatsDto` rather than `ResponseEntity<PublicStatsDto>` which is the output returned by the corresponding method in the API delegate implementation layer.